sdk/storage/Azure.Storage.Blobs/src/BlobClientOptions.cs

```diff
@@ -50,9 +56,15 @@ public enum ServiceVersion
         /// The <see cref="ServiceVersion"/> of the service API used when
         /// making requests.
         /// </param>
-        public BlobClientOptions(ServiceVersion version = LatestVersion)
+        /// <param name="customerProvidedKey">
+        /// The <see cref="CustomerProvidedKey"/> to be used when making requests.
+        /// </param>
+        public BlobClientOptions(
+            ServiceVersion version = LatestVersion,
+            CustomerProvidedKey? customerProvidedKey = default)
```

Comment:
I'm kind of split on whether this should be on the .ctor or whether it should just be a settable property.  Any idea how often this feature is used by customers?  I'm fine leaving this as-is for now and revisiting during the next detailed API review.

Reply:
Settable on BlobClientOptions vs in the BlobClientOptions constructor?